### PR TITLE
Properly parse sci-notation number like 1E-3

### PR DESCRIPTION
**Pull request by bunnymatic**
fixes issue #287

### CODE REVIEW

**Comment by stevendaniels** [comment]
Thanks!

**Comment by bunnymatic** [comment]
no problem.  glad to help out.

Do you have a schedule or plan for new releases to rubygems?  we can certainly point our gemfile to the repo, but if there will be a new release soon, we might just wait and update at that point.

let me know

**Comment by stevendaniels** [comment]
[Roo 2.3.1](https://rubygems.org/gems/roo) is out!

**Comment by bunnymatic** [comment]
awesome.  thanks

File: lib/roo/excelx/cell/number.rb [hunk 1]
```diff
@@ -15,14 +15,13 @@ def initialize(value, formula, excelx_type, style, link, coordinate)
...
           return number if Excelx::ERROR_VALUES.include?(number)
-
           case @format
           when /%/
             Float(number)
           when /\.0/
             Float(number)
           else
-            number.include?('.') ? Float(number) : Integer(number)
+            (number.include?(".") || (/\A\d+E[-+]\d+\z/ =~ number)) ? Float(number) : Integer(number)
```
Comment (Confusion) [inline lib/roo/excelx/cell/number.rb]: Hi, I locally patched this regular expression with a `-?` in front of the first `\d+`, so it also supports negative numbers in scientific notation (which I encountered, upon which I found the issue this pull request solves).

I can imagine handling a pull request is more work than just changing it yourself, but that depends on your workflow. Shall I make a pull request for it?

**Pull request merged**